hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/records/GasFeeRecordBuilder.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.contract.impl.records;
+
+public interface GasFeeRecordBuilder {
+    /**
+     * Returns the current transaction fee.
+     *
+     * @return the current transaction fee
+     */
+    long transactionFee();
+
+    /**
+     * Sets the transaction fee.
+     *
+     * @param transactionFee the new transaction fee
+     * @return the updated {@link GasFeeRecordBuilder}
+     */
+    GasFeeRecordBuilder transactionFee(long transactionFee);
+
+    /**
+     * Updates this record builder to include an additional gas fee.
+     *
+     * @param gasFee the gas fee to add
+     * @return the updated {@link GasFeeRecordBuilder}
+     */
+    default GasFeeRecordBuilder withGasFee(final long gasFee) {
```

Comment:
This fee is tinybar.  Over at `CallOutcome` the `gasPrice` is in gas.  Is that a standard use of terms "fee" == hbar units where "price" == gas?  

Regardless, perhaps mention the units in the Javadoc here and over at `CallOutcome`.

Reply:
`CallOutcome.gasPrice` is also denominated in tinybar...but agreed this is better made explicit. Will fix in [this PR](https://github.com/hashgraph/hedera-services/pull/9781) shortly; want to merge this one ASAP.